Unable to run app - missing method on Android

## 🐛 Bug Report

When trying to run my app after adding this plugin I'm getting an error with the Android plugin:

```
.pub-cache/hosted/pub.dartlang.org/flutter_pusher-1.0.1/android/src/main/java/com/ninjasolutions/pusher/PusherPlugin.java:382: error: cannot find symbol
            channel.trigger(eventName, "{}");
                   ^
  symbol:   method trigger(String,String)
  location: variable channel of type Channel
```

### Expected behavior

### Reproduction steps

### Configuration

**Version:** 1.x

**Platform:** 
- [ ] :iphone: iOS
- [X] :robot: Android

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -8,7 +8,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.0'
+        classpath 'com.android.tools.build:gradle:3.5.3'
     }
 }
 
@@ -32,6 +32,6 @@ android {
         disable 'InvalidPackage'
     }
     dependencies {
-        implementation 'com.pusher:pusher-java-client:2.0.0'
+        implementation 'com.pusher:pusher-java-client:2.0.2'
     }
 }

--- a/android/src/main/java/com/ninjasolutions/pusher/PusherPlugin.java
+++ b/android/src/main/java/com/ninjasolutions/pusher/PusherPlugin.java
@@ -13,6 +13,7 @@ import com.pusher.client.PusherOptions;
 import com.pusher.client.channel.Channel;
 import com.pusher.client.channel.ChannelEventListener;
 import com.pusher.client.channel.PresenceChannelEventListener;
+import com.pusher.client.channel.PrivateChannel;
 import com.pusher.client.channel.PrivateChannelEventListener;
 import com.pusher.client.channel.PusherEvent;
 import com.pusher.client.channel.User;
@@ -377,9 +378,18 @@ public class PusherPlugin implements MethodCallHandler {
             final String channelName = json.getString("channelName");
             final String eventName = json.getString("eventName");
 
-            Channel channel = channels.get(channelName);
+            String data = "{}";
+            if (json.has("data")) {
+                data = json.getString("data");
+            }
 
-            channel.trigger(eventName, "{}");
+            Channel channel = channels.get(channelName);
+            if (channel instanceof PrivateChannel) {
+                if (isLoggingEnabled) {
+                    Log.d(TAG, String.format("Trigger event: %s with data: %s", eventName, data));
+                }
+                ((PrivateChannel) channel).trigger(eventName, data);
+            }
         } catch (Exception e) {
             if (isLoggingEnabled) {
                 Log.d(TAG, String.format("unbind exception: %s", e.getMessage()));

